source/Irrlicht/CImageLoaderBMP.cpp

```diff
@@ -155,15 +162,16 @@ void CImageLoaderBMP::decompress4BitRLE(u8*& bmpData, s32 size, s32 width, s32 h
 					s32 count = (u8)*p; ++p;
 					s32 readAdditional = ((2-((count)%2))%2);
 					s32 readShift = 4;
-					s32 i;
 
-					for (i=0; i<count; ++i)
+					CHECKP(SHIFTED_COUNT(count, readShift));
+					CHECKD(SHIFTED_COUNT(count, shift));

```

Comment:
What does this check?

Reply:
check that `p`/`d` are big enough for which bytes the following loop touches